py-polars/tests/unit/dataframe/test_df.py

```diff
@@ -3253,22 +3253,24 @@ def test_item() -> None:
     assert df.item() == 1
 
     df = pl.DataFrame({"a": [1, 2]})
-    with pytest.raises(ValueError):
+    with pytest.raises(ValueError, match=r".* frame has shape \(2, 1\)"):
         df.item()
 
     assert df.item(0, 0) == 1
     assert df.item(1, "a") == 2
 
     df = pl.DataFrame({"a": [1], "b": [2]})
-    with pytest.raises(ValueError):
+    with pytest.raises(ValueError, match=r".* frame has shape \(1, 2\)"):
         df.item()
 
     assert df.item(0, "a") == 1
     assert df.item(0, "b") == 2
 
     df = pl.DataFrame({})
-    with pytest.raises(ValueError):
+    with pytest.raises(ValueError, match=r".* frame has shape \(0, 0\)"):
         df.item()
+    with pytest.raises(ValueError, match='Column index "10" is out of bounds'):
```

Comment:
Shouldn't we return IndexError?

Reply:
We should, but I didn't want `item` to diverge from `__getitem__`, where a `ValueError` is currently being raised. 

Want me to update both locations to `IndexError` instead? I suppose _technically_ that would be a breaking change, but it's probably a stretch to call it such  ;)

Reply:
I'd say keep the ValueError here, and update to IndexError in a separate PR. We can decide if we want to break then.